tests/test_search.py

```diff
@@ -1,67 +1,200 @@
 #!/usr/bin/env python
 # coding: utf-8
 
+import datetime
 import json
 import logging
 import os
 import sys
+import time
 import unittest
+import uuid
 
 import requests
 
-pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))  # noqa
 sys.path.insert(0, pkg_root)  # noqa
 
 import dss
+from dss import DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE
 from dss.events.handlers.index import create_elasticsearch_index
-from dss.util.es import ElasticsearchServer
-from tests.infra import DSSAsserts
+from dss.util.es import ElasticsearchServer, ElasticsearchClient
+from tests.infra import DSSAsserts, start_verbose_logging
+from tests.es import elasticsearch_delete_index
+from tests import IndexSearchTestSupport
 
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
 
-class TestSearch(unittest.TestCase, DSSAsserts):
+start_verbose_logging()
+
+class TestSearch(unittest.TestCase, DSSAsserts, IndexSearchTestSupport):
     @classmethod
     def setUpClass(cls):
         cls.es_server = ElasticsearchServer()
         os.environ['DSS_ES_PORT'] = str(cls.es_server.port)
+        dss.Config.set_config(dss.DeploymentStage.TEST)
+        cls.app = dss.create_app().app.test_client()
+        cls.index_document = cls.load_sample_document("sample_index_doc.json")
+
+    @classmethod
+    def load_sample_document(cls, file_name):
+        with open(os.path.join(os.path.dirname(__file__), file_name), "r") as fh:
+            return json.load(fh)
 
     @classmethod
     def tearDownClass(cls):
         cls.es_server.shutdown()
 
     def setUp(self):
-        dss.Config.set_config(dss.DeploymentStage.TEST)
-        self.app = dss.create_app().app.test_client()
-
-        create_elasticsearch_index(logging.getLogger(__name__))
+        elasticsearch_delete_index("_all")
```

Comment:
Since both create and delete index operations must wait for the primary shard within elasticsearch, the operations should happen synchronously.

https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html
https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-index.html
https://discuss.elastic.co/t/race-condition-when-removing-indexing-and-counting-data-by-using-the-java-api/2190